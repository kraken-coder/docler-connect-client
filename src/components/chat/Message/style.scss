@import '../../../template/layout.scss';

.msg-wrapper {
  margin: 2rem 0.5rem 0 0.5rem;
  display: flex;
  flex-direction: column;
  background: none;
  padding: 1rem 1.5rem;
  width: 80%;
  .msg-top {
    color: set-text-color(color(primary));
    margin-bottom: 0.5rem;
    .msg-user {
      font-size: 1rem;
    }
  }
  .msg-bottom {
    margin-top: 0.5rem;
    .msg-time {
      font-size: 1rem;
      color: set-text-color(color(primary));
    }
  }
  .msg-container {
    background: color(transparent);
    .msg {
      font-size: 1rem;
      color: color(primary);
    }
  }
}
.sentContainer {
  float: right;
}
.recievedContainer {
  float: left;
}

.sent {
  color: set-text-color(color(transparent));
  padding: 1rem 1.5rem;
  border-top-left-radius: 4rem;
  border-top-right-radius: 4rem;
  border-bottom-right-radius: 4rem;
}

.recieved {
  color: set-text-color(color(transparent));
  border-top-left-radius: 4rem;
  border-top-right-radius: 4rem;
  border-bottom-left-radius: 4rem;
  padding: 1rem 1.5rem;
  // clip-path: polygon(0% 0%, 100% 0%, 100% 75%, 45% 74%, 26% 100%, 19% 75%, 0% 75%);
}
