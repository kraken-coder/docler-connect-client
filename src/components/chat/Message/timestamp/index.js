import TimeStamp from './TimeStamp'

export default TimeStamp
