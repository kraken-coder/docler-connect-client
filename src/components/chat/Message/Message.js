/* eslint-disable react/prop-types */
import React from 'react'
import { connect } from 'react-redux'
// import PropTypes from 'prop-types'

//  core components
import User from './user'
import TimeStamp from './timestamp'
import './style.scss'

const Message = ({
  message: { user, text },
  settings: { clockMode },
  chat: { username },
}) => {
  //  check to sse if message is from the current user
  let isSentByUser = false
  if (username === user) {
    isSentByUser = true
  }
  //  display message based on the outcome of the function above
  return isSentByUser ? (
    <>
      <div className="msg-wrapper SentContainer ">
        <div className="msg-container sent">
          <p className="msg">{text}</p>
        </div>
        <TimeStamp clockMode={clockMode} />
      </div>
    </>
  ) : (
    <>
      <div className="msg-wrapper recievedContainer">
        <User user={user} />
        <div className="msg-container recieved">
          <p className="msg">{text}</p>
        </div>
        <TimeStamp clockMode={clockMode} />
      </div>
    </>
  )
}

const mapStateToProps = state => ({
  settings: state.settings,
  chat: state.chat,
})

const exports = React.memo(Message)
export default connect(mapStateToProps)(exports)
