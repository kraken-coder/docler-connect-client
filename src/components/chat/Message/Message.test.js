import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import toJSON from 'enzyme-to-json'
import Message from './Message'

// mock state
const initialState = {}
const mockStore = configureStore()
const store = mockStore(initialState)

test('match snapshot', () => {
  const wrapper = shallow(<Message store={store} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
