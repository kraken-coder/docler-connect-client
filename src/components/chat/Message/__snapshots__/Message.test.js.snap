// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Memo(Message)
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
